graphrag_eval/steps/retrieval_evaluation_using_context_texts.py

```diff
@@ -0,0 +1,82 @@
+from langevals_ragas.context_precision import (
+    RagasContextPrecisionEntry,
+    RagasContextPrecisionEvaluator,
+)
+from langevals_ragas.context_recall import (
+    RagasContextRecallEntry,
+    RagasContextRecallEvaluator,
+)
+
+
+def _evaluate(
+    entry: RagasContextRecallEntry | RagasContextPrecisionEntry,
+    evauator: RagasContextRecallEvaluator | RagasContextPrecisionEvaluator,
+    metric: str
+) -> dict:
+    try:
+        result = evauator.evaluate(entry)
+        print(type(result))
+        print(result)
+        print(result.cost)
+        if result.status == "processed":
+            result_dict = {
+                f"retrieval_{metric}": result.score,                
+            }
+            if result.details:
+                result_dict[f"retrieval_{metric}_reason"] = result.details
+            if result.cost is not None:
+                result_dict[f"retrieval_{metric}_cost"] = result.cost.amount
+            return result_dict
+        else:
+            return {
+                f"retrieval_{metric}_error": result.details,
+            }
+    except Exception as e:
+        return {
+            f"retrieval_{metric}_error": str(e),
+        }
+
+def get_f1_dict(
+    input_dict: dict[str, float | str],
+) -> dict[str, float | str]:
+    recall = input_dict.get("retrieval_recall")
+    precision = input_dict.get("retrieval_precision")
+    result = {}
+    if recall is not None and precision is not None:
+        if precision == 0 or recall == 0:
```

Comment:
In the file retrieval_answer.py we have a similar function, the f1 score is calculated almost the same as here, these we have ``if recall == 0.0 and precision == 0.0``. Do you think we can refactor the code, so that we have one version of this function, or at least the part with the f1 calculation